akka-stream/src/main/scala/akka/stream/impl/io/TcpConnectionStream.scala

```diff
@@ -91,7 +89,6 @@ private[akka] abstract class TcpStreamActor(val settings: ActorMaterializerSetti
       if (!closed) {
         closed = true
         pendingElement = null
-        if (!tcpOutputs.isFlushed && (connection ne null)) connection ! ResumeReading
```

Comment:
I don't remember why that condition is there, but `isFlushed` means that the TCP write side has been fully closed with no writes/bytes pending.